src/grammar-registry.coffee

```diff
@@ -57,15 +55,17 @@ class GrammarRegistry extends FirstMate.GrammarRegistry
     return -1 unless filePath
     filePath = filePath.replace(/\\/g, '/') if process.platform is 'win32'
 
-    pathComponents = filePath.toLowerCase().split(PathSplitRegex)
+    pathComponents = filePath.toLowerCase().split('/')
+    pathComponents = pathComponents.concat(pathComponents.pop().split('.'))
```

Comment:
I just implemented this, but then realized we had a previous discussion about it: https://github.com/atom/atom/pull/10145#discussion-diff-52680158.  Is this something that we're sure won't run into `file:///` or other weird Windows normalization difficulties?